app/services/state_file/state_information_service.rb

```diff
@@ -1,3 +1,4 @@
+# frozen_string_literal: true
```

Comment:
can probably take out this note

Reply:
Thanks for the detailed look! This isn't actually a note, but a "magic comment" (sometimes called a "pragma") that Rubymine puts in new ruby files (and which we should probably have more consistently throughout our codebase).

When this comment is at the top of the file, it tells Ruby that the value of the string literals in that file (anything in single or double quotes) can not be changed by other code. In this case, it's not so critical since we use "deep_freeze" on the whole `STATES_INFO` dictionary, but it can matter in some cases both for performance and to make sure that strings are not changed unintentionally while the program is running. Happy to talk more about it if you're interested - it's pretty low-level/nuanced stuff that doesn't usually matter. Here's a document that goes into a bit more detail: https://gist.github.com/fxn/bf4eed2505c76f4fca03ab48c43adc72